src/compo/tropomi_no2_nc2ioda.py

```diff
@@ -60,111 +52,131 @@ def __init__(self, filenames):
     # Open input file and read relevant info
     def _read(self):
         # set up variable names for IODA
-        for iodavar in ['nitrogen_dioxide_in_tropospheric_column', 'nitrogen_dioxide_in_total_column']:
-            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
-            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
-            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
-            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
-            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
-            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
-            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'mol m-2'
-            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'mol m-2'
+        if self.columnType == 'total':
+            iodavar = self.obsVar['nitrogendioxide_total_column']
+        elif self.columnType == 'tropo':
+            iodavar = self.obsVar['nitrogendioxide_tropospheric_column']
+        self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+        self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+        self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+        self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+        self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+        self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+        self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'mol m-2'
+        self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'mol m-2'
         # loop through input filenames
         first = True
         for f in self.filenames:
             ncd = nc.Dataset(f, 'r')
+
             # get global attributes
             AttrData['date_time_string'] = ncd.getncattr('time_reference')[0:19]+'Z'
             AttrData['sensor'] = ncd.getncattr('sensor')
             AttrData['platform'] = ncd.getncattr('platform')
+
             # many variables are time, scanline, ground_pixel
             # but others are just time, scanline
             lats = ncd.groups['PRODUCT'].variables['latitude'][:].ravel()
             lons = ncd.groups['PRODUCT'].variables['longitude'][:].ravel()
             qa_value = ncd.groups['PRODUCT'].variables['qa_value'][:]  # 2D
             times = np.empty_like(qa_value, dtype=object)
             qa_value = qa_value.ravel()
+
+            # adding ability to pre filter the data using the qa value
+            # documentation recommends using > 0.75
```

Comment:
Is this the documentation from the data provider? If so it might be good to include, for the next developer, a URL reference to where the recommendation exists.

Reply:
> Thanks for the update and for your patience with my slow response. I've just got the one comment that I am viewing as optional.

No problems, thanks a lot for the review!

Reply:
I will put the url of the documentation on the next PR on this.